styx-api-service/src/main/java/com/spotify/styx/api/CliResource.java

```diff
@@ -96,11 +94,21 @@ public CliResource(String schedulerServiceBaseUrl, EventStorage eventStorage) {
 
     final List<Route<AsyncHandler<Response<ByteString>>>> proxies = Arrays.asList(
         Route.async(
-            "POST", BASE + "/events",
-            this::injectEventProxy),
+            "GET", BASE + "/<endpoint>",
+            rc -> proxy("/" + arg("endpoint", rc), rc)),
         Route.async(
-            "POST", BASE + "/trigger",
-            this::triggerWorkflowInstanceProxy));
+            "POST", BASE + "/<endpoint>",
+            rc -> proxy("/" + arg("endpoint", rc), rc)),
+        Route.async(
+            "GET", BASE + "/<endpoint>/<id>",
+            rc -> proxy("/" + arg("endpoint", rc) + "/" + arg("id", rc), rc)),
+        Route.async(
+            "DELETE", BASE + "/<endpoint>/<id>",
+            rc -> proxy("/" + arg("endpoint", rc) + "/" + arg("id", rc), rc)),
+        Route.async(
+            "PATCH", BASE + "/<endpoint>/<id>",
+            rc -> proxy("/" + arg("endpoint", rc) + "/" + arg("id", rc), rc))
```

Comment:
You might want to replace these all with just a `/<endpoint:path>` parameter that will match the rest of the uri. see https://github.com/spotify/apollo/blob/master/apollo-api/docs/routing-engine.md#path-parameters